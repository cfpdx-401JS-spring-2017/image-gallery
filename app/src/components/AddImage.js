import React, { Component } from 'react';

export default class AddImage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      description: '',
      url: ''
    };

    this.handleChange = this.handleChange.bind(this);
  }

  reset() {
    this.setState({
      name: '',
      description: '',
      url: ''
    });
  }

  handleChange(event) {
    const { name, value } = event.target;
    this.setState({
      [name]: value,
    });
  }

  render() {
    const { name, description, url } = this.state;

    return (
      <div>
        <h4>Add Your Own Poms!</h4>
        <form onSubmit={e => {
          e.preventDefault();
          this.props.onAdd(this.state);
          this.reset();
        }}>
          <label>
            name: <input name="name"
              required
              value={name}
              onChange={this.handleChange} />
          </label>
          <label>
            url: <input name="url"
              required
              value={url}
              onChange={this.handleChange} />
          </label>
          <label>
            description: <input name="description"
              required
              value={description}
              onChange={this.handleChange} />
          </label>
          <button type="submit">Add</button>
        </form>
      </div>
    );
  }
}