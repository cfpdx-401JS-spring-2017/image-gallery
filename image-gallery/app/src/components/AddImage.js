import React, { Component } from 'react';

export default class AddImage extends Component {
    //passing onAdd Handler from UpdateImages.js
    constructor(props) {
        super(props);

        this.state = {
            title: '',
            description: '',
            url: '',
            wiki: '',
        };

        this.handleChange = this.handleChange.bind(this);
    }

    reset() {
        this.setState({
            title: '',
            description: '',
            url: '',
            wiki: ''
        })
    }

    //Q:are we saying let the value of the input become the name?
    handleChange(event) {
        const { name, value } = event.target;
        this.setState({
            [name]: value
        });
    }

    //Q: Why are we passing this.state in line 43 when the fn takes (image)?
    render() {
        const { title, description, url, wiki } = this.state;

        return (
            <div className="add-image">
                <h2>Add An Image:</h2>
                <form onSubmit={event => {
                    event.preventDefault();
                    this.props.handleAdd(this.state);
                    this.reset();
                }}>
                    <label>
                        Title: <input name="title"
                            required
                            value={title}
                            onChange={this.handleChange} />
                    </label>
                    <label>
                        Description: <input name="description"
                            value={description}
                            onChange={this.handleChange} />
                    </label>
                    <label>
                        Image URL: <input name="url"
                            required
                            value={url}
                            onChange={this.handleChange} />
                    </label>
                    <label>
                        Link to More Info: <input name="wiki"
                            value={wiki}
                            onChange={this.handleChange} />
                    </label>
                    <button type="submit">Add Image</button>
                </form>

            </div>
        )
    }




}