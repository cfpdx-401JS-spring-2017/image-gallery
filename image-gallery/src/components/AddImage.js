import React, { Component } from 'react';

export default class AddImage extends Component {
    //passing onAdd Handler from UpdateImages.js
    constructor(props) {
        super(props);

        this.state = {
            title: '',
            description: '',
            url: '',
            wiki: '',
        };

        this.handleChange = this.handleChange.bind(this);
    }

    reset() {
        this.setState({
            title: '',
            description: '',
            url: '',
            wiki: ''
        })
    }

    handleChange(event) {
        const { title, description, url, wiki } = event.target;
        this.setState({
            [title]: value
        });
    }
    render() {
        const { title, description, url, wiki } = this.state;

        return (
            <div>
                <h2>Add An Image:</h2>
                <form onSubmit={event => {
                    event.preventDefault();
                    this.props.onAdd(this.state);
                    this.reset();
                }}>
                    <label>
                        Title: <input name="title"
                            required
                            value={title}
                            onChange={this.handleChange} />
                    </label>
                </form>
            </div>
        )
    }




}