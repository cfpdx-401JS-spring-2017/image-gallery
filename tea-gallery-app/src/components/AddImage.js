import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AddImage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      title: '',
      description: '',
      url: ''
    };
    
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;
    this.setState({ [name]: value });
  }

  render() {
    return (
      <form className="add-item" onSubmit={e => {
        e.preventDefault();
        this.props.onAddImage(this.state);
      }}>
        <label>Image name<input name='title' onChange={this.handleChange}></input></label><br/>
        <label>Description<input name='description' onChange={this.handleChange}></input></label><br/>
        <label>URL<input name='url' onChange={this.handleChange}></input></label><br/>
        <button type="submit">Add</button>
      </form>
    );
  }
}

AddImage.propTypes = {
  onAddImage: PropTypes.func,
};

export default AddImage;