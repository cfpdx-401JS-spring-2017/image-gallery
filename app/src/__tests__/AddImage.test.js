import React from 'react';
import renderer from 'react-test-renderer';
import AddImage from '../AddImage';

describe('AddImage', () => {
  it('renders a form to add images', () => {
    const tree = renderer.create(<AddImage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});