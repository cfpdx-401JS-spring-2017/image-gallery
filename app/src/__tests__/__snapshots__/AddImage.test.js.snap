// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddImage renders a form to add images 1`] = `
<div
  className="Add-image"
>
  <h2>
    Add Image
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label>
      Title:
      <input
        onChange={[Function]}
        value=""
      />
    </label>
    <label>
      Description:
      <input
        onChange={[Function]}
        value=""
      />
    </label>
    <label>
      URL:
      <input
        onChange={[Function]}
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
